db/aliasManager.go

```diff
@@ -0,0 +1,242 @@
+package db
+
+import (
+	"context"
+	"strings"
+	"strconv"
+	"errors"
+	"fmt"
+	"math/rand"
+	tinycrypt "github.com/uclaacm/teach-la-go-backend-tinycrypt" 
+	"cloud.google.com/go/firestore"
+	"google.golang.org/api/iterator"
+)
+
+type Shard struct {
+	Count int64
+}
+
+var crypt = tinycrypt.Encrypter{
+	[]uint64 {
+		0x33684192D,
+		0x28DAB6A5A,
+		0xA928A9246,
+		0x224D23A42,
+	},
+}
+
+
+
+
+func (d *DB) InitShards(ctx context.Context, path string)(error){
+
+	// get the document that has the counters
+	doc := d.Collection(path).Doc("--shards--")
+
+	// get the subcollection with the sub-counters
+	col := doc.Collection("shards")
+
+	// Initialize each to 0
+	for i := 0; i < num_shards; i++{
+
+		d :=  col.Doc(strconv.Itoa(i))
+		fmt.Print(d.ID)
+		_, err := col.Doc(strconv.Itoa(i)).Set(ctx, map[string]interface{}{
+			"Count" : 0,
+		})
+		
+		if err != nil {
+			return err
+		}
+	} 
+
+	// en.InitializeEncrypter(keys)
+
+	return nil
+
+}
+
+func (d *DB) GetID(ctx context.Context, path string)(int64, error){
+
+	// get the document that has the counters
+	doc := d.Collection(path).Doc("--shards--")
+
+	// get every document in the collection
+	shards := doc.Collection("shards").Documents(ctx)
+	count := make([]int64, num_shards)
+
+	// check each shard 
+	i := 0
+	for {
+
+		d, e := shards.Next()
```

Comment:
Should probably put your incrementation at the bottom of the loop to ensure you reach all documents, but there may be a detail I'm missing here.

Reply:
This was how they did it in the official tutorial, tbh I'm not confident enough to challenge professionals at Firebase. https://firebase.google.com/docs/firestore/solutions/counters

Reply:
Probably all clear then. Actually makes sense on second thought now that I see that `d` and `e` are both initialized here.